src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -5091,6 +5092,34 @@ RoundRobinReorder(Task *task, List *placementList)
 }
 
 
+
+List *
+RoundRobinPerTransactionAssignTaskList(List *taskList)
+{
+	taskList = ReorderAndAssignTaskList(taskList, RoundRobinPerTransactionReorder);
+
+	return taskList;
+}
+
+/*
+ * RoundRobinPerTransactionReorder implements the core of the round-robin assignment policy.
+ * It takes a task and placement list and rotates a copy of the placement list
+ * based on the task's jobId. The rotated copy is returned.
+ */
+static List *
+RoundRobinPerTransactionReorder(Task *task, List *placementList)
+{
+	TransactionId transactionId = GetStableLatestTransactionId();
```

Comment:
I don't want to rely on distributed transactionId since it is only set on the execution. And, changing that would probably end up with lots of edge cases.

Reply:
Makes sense.